### ISSUE TITLE
Weird jumping around image on component page

### ISSUE THREAD

**Issue by kevmoo** [dart-lang/site-webdev#1018]
On https://webdev.dartlang.org/components

See https://youtu.be/55ZpJ9o5bdI

**Comment by kevmoo** [comment]
Same here! https://webdev.dartlang.org/angular

**Comment by Sfshaza** [comment]
That is weird!

**Comment by kwalrath** [comment]
Funny, I used to see that all the time, and now I don't any more. Maybe because I'm using Chrome canary (v62)?

/cc @chalin

**Comment by chalin** [comment]
Yes, this is a known problem (though I thought we already had an issue open over it).

This problem is a consequence of how the sidenav was originally implemented quite a while ago.
IMHO, the sidenav, and the autogenerated toc implementations need an overhaul.

I'd found a fix/workaround we few weeks ago, but didn't have time to fully test it out.

**Comment by chalin** [comment]
Btw, what is the preference here? That I patch it on a per file basis, or that some time be invested in looking at a better sidenav (and possibly new autogenerated toc) implementation(s)?

**Comment by kwalrath** [comment]
@filiph Do you have any guidance here? I'm inclined to look into better sidenav/toc (which could be used on www.dartlang.org, too), but I don't want to open a can of worms...

**Comment by chalin** [comment]
We don't have to make a go/no-go decision yet. I can do some preliminary research and come back with concrete recommendations (and a possible timeline).

**Comment by filiph** [comment]
Yes, I agree looking into a more systemic approach is better. As much as I hate the glitch, I don't think we should spend time chasing it around and squatting it case-by-case.

The sidenav was implemented by our agency as a [bunch of jquery hacks](https://github.com/dart-lang/site-webdev/blob/master/src/_assets/javascripts/main.js#L12). Not complaining here, it was a pragmatic decision at the time, just saying that we can do better. My long-standing wish is to get rid of the jquery and use simple Dart for this. (For that, we'd need to find or re-implement jQuery's `animate` and `smartresize` — the rest is very 1:1 with dart:html. And of course we'd need to come up with a good way to build Dart-to-JS as part of a Jekyll build, which is surprisingly hard the last time I tried.)

I digress! Let's see if we can fix the jumping with a simple yet systemic fix in JS.

**Comment by chalin** [comment]
Kathy might have mentioned this before, but in any case: the sidenav and toc (right-floating when wide enough, inlined at the top of the page otherwise) on https://developers.google.com/style/ are very nice.

**Comment by kwalrath** [comment]
I filed #868 about this. Someone volunteered to fix it a month ago, but I haven't heard from them.

**Comment by chalin** [comment]
Folding #138 into this issue.

**Comment by chalin** [comment]
I've edited the title so that this issue covers only the sidenav (since the TOC doesn't affect the originally mentioned issue of images jumping on the page). #868 will be used to track TOC related improvements.

**Comment by chalin** [comment]
FYI, the firebase site sidenav is quite nice, and it supports up to 3 (or 4 depending on how you count) levels. [E.g.](https://firebase.google.com/docs/hosting/functions):
![image](https://user-images.githubusercontent.com/4140793/31574986-a6440428-b0a9-11e7-911b-2b722ce77160.png)
